CACI-MIlMOVE/mymove
# B-17235 Okta API Call to Edit

## [B-17235 Okta API Call to Edit](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A834295&RoomContext=TeamRoom%3A808731&concept=TeamRoom)

## Summary

When a user goes to make changes to their Okta profile, they input updated values into the form.
Upon clicking "Save", this will call the Okta Users API and update their profile.
MilMove will then reflect changed values in state as well as calling a GET request to the Okta API on page refresh to ensure updated values are shown and not values in Access Token (which can be dated).

## Verification Steps for Reviewers

Upon logging into Customer MilMove, please ensure your Okta profile values are up to date and accurate.
Saving new values in the form should send you back to the profile page, display a success banner, and you should see the updated values in the form viewer.

### Setup to Run the Code

- switch to branch
- `make client_run`
- `make server_run`
- Once built and compiled, log into MilMove using your Okta credentials as a customer (please note that if you are a part of the office OR admin groups, you will not be able to access the application and will need to create a new user or remove yourself from those groups in Okta).
- Navigate to the profile section
- Update your Okta values
- Ensure changes are accurate

### Frontend

- Updated EditOktaInfo component to reflect Okta values stored in state
- Updated Redux store with Okta actions that are fired when `SET_OKTA_USER` action is called
- Reflected changes in swagger UI and .yaml files

### Backend

- Added GET and POST requests to `okta_profile.go` handler
